src/SW_Weather.c

```diff
@@ -1124,11 +1575,28 @@ void SW_WTH_read(void) {
     @param year Current year within the simulation
     @param yearWeather Current year's weather array that is to be filled by function
     @param weather_prefix File name of weather data without extension.
+    @param n_input_forcings Number of read-in columns from disk
+    @param dailyInputIndices An array of size MAX_INPUT_COLUMNS holding the calculated
+    column number of which a certain variable resides
+    @param dailyInputFlags An array of size MAX_INPUT_COLUMNS holding booleans specifying
+    what variable has daily input on disk
+    @param use_cloudCoverMonthly A boolean; specifying whether or not cloud cover monthly values
+    are to be used instead of daily input values
+    @param use_windSpeedMonthly A boolean; specifying whether or not wind speed monthly values
+    are to be used instead of daily input values
+    @param use_humidityMonthly A boolean; specifying whether or not humidity monthly values
+    are to be used instead of daily input values
 */
 void _read_weather_hist(
   TimeInt year,
   SW_WEATHER_HIST *yearWeather,
-  char weather_prefix[]
+  char weather_prefix[],
+  unsigned int n_input_forcings,
+  unsigned int *dailyInputIndices,
+  Bool *dailyInputFlags,
+  Bool use_cloudCoverMonthly,

```

Comment:
I think we should get rid of the three monthly use flags from `_read_weather_hist()`
- this function is about daily values
- `SW_WTH_setup()` turns off daily use flags if monthly use flags are turned on -- making the information here redundant

Reply:
I meant to mention this to you. I think it would be best to remove cloud cover and wind speed monthly flags. I think it would be a good idea, unless argued otherwise, to keep `use_humidityMonthly`. I think this because it's more efficient to skip the entire code block dealing with humidity values. I.e., instead of having all daily flags be turned off and check these flags ~11 times a day instead of one (`use_humidityMonthly).

Reply:
Good point considering how many logical conditions would be tested daily -- I didn't think about that aspect! I still think we should remove all three monthly use flags to clean up the interface; what do think about creating a local variable `use_humidityDaily` that is calculated from the daily flags and which we can use instead of `!use_humidityMonthly`?

Reply:
I think that would be a good idea since it would, like you stated, make the interface easier to deal with. I think your idea is the best. I just need to see if keeping it static would be a good idea. I don't believe so since the initial testing and setting of the value would be difficult due to the fact that we cannot use SW_MISSING, since it's of type Bool. It's possible to make it static but may require more than it's worth. The comparison of number of checks in 31 years would be ~68,076 > ~186 > ~6 (completely getting rid of monthly humidity flag, using your idea as non-static, using your idea as a static variable, respectively).

Though, am I mistaken, or if we are to solely calculate if `use_humidityMonthly` from other flags, is there not a chance a user can turn off the monthly flags and have none of the subsequent daily flags set, so the program assumes that the monthly flag is in use? This should just result in a crash within `SW_WTH_new_day()`, correct?